sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/subquery.scala

```diff
@@ -248,7 +248,10 @@ case class ScalarSubquery(
     children: Seq[Expression] = Seq.empty,
     exprId: ExprId = NamedExpression.newExprId)
   extends SubqueryExpression(plan, children, exprId) with Unevaluable {
-  override def dataType: DataType = plan.schema.fields.head.dataType
+  override def dataType: DataType = {
+    assert(!plan.schema.fields.isEmpty, "scalar subquery does not have any column")
```

Comment:
Can we use `plan.schema. fields.nonEmpty` instead?

Reply:
Also, the message looks a bit odd. It should rather be `Scala subquery should have only one column`